tests/helpers_for_tests.py

```diff
@@ -0,0 +1,38 @@
+import numbers
+
+import numpy as np
+import pytest
+from pytest import approx
+
+
+def compare_dicts(d_ref, d):
```

Comment:
I am using dictdiffer library to compare dictionaries (see test_pipeline_output.py) for this. Is there a reason why you would use a custom function instead?

Reply:
@sgratiy Thanks for the pointer. I was not aware of dictdiffer and my first tests are not promising unfortunately.

There is a bug in dictdiffer as it can not handle numpy datatypes properly as

```
In [58]: list(diff({"a": np.float32('nan')}, {"a" : float('nan')}))
Out[58]: [('change', 'a', (nan, nan))]

In [60]: are_different(np.float32('nan'),np.nan, 0)
Out[60]: True
```

shows. We get the np.float32('nan') value from the NWB file.

I'll file a bug report with them ASAP.

Reply:
I've documented in 7e770b2d (Tests/compare_dicts: Explain why we don't use dictdiffer here, 2018-12-17) why dictdiffer does not work.